xrpl4j-client/src/main/java/org/xrpl/xrpl4j/client/XrplClient.java

```diff
@@ -135,13 +141,49 @@ public XrplClient(HttpUrl rippledUrl) {
     return jsonRpcClient.send(request, resultType);
   }
 
+  /**
+   * Submit a {@link SignedTransaction} to the XRP Ledger.
+   *
+   * @param <T>               The type of signed {@link Transaction} that is being submitted.
+   * @param signedTransaction A {@link org.xrpl.xrpl4j.crypto.signing.SignedTransaction} to submit.
+   *
+   * @return The {@link SubmitResult} resulting from the submission request.
+   *
+   * @throws JsonRpcClientErrorException If {@code jsonRpcClient} throws an error.
+   * @throws JsonProcessingException     if any JSON is invalid.
+   * @see "https://xrpl.org/submit.html"
+   */
+  public <T extends Transaction> SubmitResult<T> submit(
+    final org.xrpl.xrpl4j.crypto.signing.SignedTransaction signedTransaction
+  ) throws JsonRpcClientErrorException, JsonProcessingException {
+
+    if (LOGGER.isDebugEnabled()) {
+      LOGGER.debug("About to submit signedTransaction: {}", signedTransaction);
+    }
+
+    String signedJson = objectMapper.writeValueAsString(signedTransaction.signedTransaction());
+    String signedBlob = binaryCodec.encode(signedJson); // <-- txBlob must be binary-encoded.
```

Comment:
Move this into `SignatureService` and add `signedBlob` to `SignedTransaction`. Also populate `Transaction.hash()` inside of `SignatureService` inside the `SignedTransaction`

Reply:
@nkramer44 This is moved into the `SigningService`, although I decided to maintain the original pattern that you introduced in `main` a few days ago because after I experimented with the design we discussed during PR review, I concluded that your original design was superior -- so, if a developer wants to get a "pre-handle" `hash` before submitting to the XRPL, the developer can use the `hash` value on `SignedTransaction`. This _should_ match the hash value on any transaction returned from XRPL via `account_tx` call.

Take a look here to see if you agree: https://github.com/XRPLF/xrpl4j/commit/2ee265f072626706d5920ac8dc52569b7c484c3b

Reply:
Yep this looks good to me